### ISSUE TITLE
Single column text encoder may need to bypass mixer

### ISSUE THREAD

**Issue by hakunanatasha** [mindsdb/lightwood#423]
The text encoder adjusts the representation of the embedding with the target variable (currently as of 2021.04.23, only classification tunes the embedding's representation). Annoyingly, the encoder's logit output is actually better at classifying language than the mixer, unsurprisingly because the training procedure involves the transformer architecture whereas the mixer doesn't update with this in mind.

This is only a concern when there is a single column of text (i.e. IMDB reviews dataset). Some following proposals:

* For single-column, text only, inherit the encoder.classifier layer as the mixer
* Create a special mode to flag the output of the encoder.classifier layer to be returned
* Create an identity mixer with the goal of exactly outputting the encoder's logit-layer
* Joint train mixer + encoder (this is not ideal, as parallelizing the encoder training would be ideal)

These are some suggestions. Currently, the benchmarks achieve SOTA, but the mixer does not "understand" language as the encoder can (as indicated by an internal test with generated reviews versus the test set).

**Comment by George3d6** [comment]
@hakunanatasha do you remember the result of our discussion vis-a-vis this issue? I remember we spoke about it but I honestly forgot.

**Comment by hakunanatasha** [comment]
The order listed above is roughly the order we thought was "most easily feasible" to "least". 

In particular, I think "Create a special mode to flag the output of the encoder.classifier layer to be returned" was the fastest to implement, albeit not the prettiest. Let me know if I'm missing something.

**Comment by George3d6** [comment]
done as part of new lightwood
